app/src/gui/pages/record-create.tsx

```diff
@@ -230,13 +233,16 @@ function DraftRecordEdit(props: DraftRecordEditProps) {
             variant={is_mobile ? undefined : 'outlined'}
           >
             {is_link_ready ? (
-              <InheritedDataComponent
-                parentRecords={parentLinks}
-                ui_specification={uiSpec}
-              />
+              checkIfParentHasInheritedData({parentLinks, uiSpec}) ? (
```

Comment:
I'd suggest moving this test inside InheritedDataComponent rather than requiring this construct here.